contracts/registry/ReverseRegistrar.sol

```diff
@@ -18,55 +23,82 @@ contract ReverseRegistrar {
      * @param ensAddr The address of the ENS registry.
      * @param resolverAddr The address of the default reverse resolver.
      */
-    constructor(ENS ensAddr, NameResolver resolverAddr) public {
+    constructor(ENS ensAddr, NameResolver resolverAddr) {
         ens = ensAddr;
         defaultResolver = resolverAddr;
 
         // Assign ownership of the reverse record to our deployer
-        ReverseRegistrar oldRegistrar = ReverseRegistrar(ens.owner(ADDR_REVERSE_NODE));
+        ReverseRegistrar oldRegistrar = ReverseRegistrar(
+            ens.owner(ADDR_REVERSE_NODE)
+        );
         if (address(oldRegistrar) != address(0x0)) {
             oldRegistrar.claim(msg.sender);
         }
     }
 
+    modifier authorised(address addr) {
+        require(
+            controllers[msg.sender] ||
+                addr == msg.sender ||
```

Comment:
For gas-efficiency, we probably want to put this one first.

Reply:
Is this because the lookup on `controllers[msg.sender]` is more costly than a simple equality?